programs/uxd/src/instructions/credix_lp/rebalance_request_execute_from_credix_lp_depository.rs

```diff
@@ -0,0 +1,646 @@
+    /// #24
+    pub associated_token_program: Program<'info, AssociatedToken>,
+    /// #25
+    pub credix_program: Program<'info, credix_client::program::Credix>,
+    /// #26
+    pub rent: Sysvar<'info, Rent>,
+}
+
+pub(crate) fn handler(ctx: Context<RebalanceRequestExecuteFromCredixLpDepository>) -> Result<()> {
+    let credix_global_market_state = ctx.accounts.depository.load()?.credix_global_market_state;
+    let collateral_mint = ctx.accounts.depository.load()?.collateral_mint;
+    let depository_pda_signer: &[&[&[u8]]] = &[&[
+        CREDIX_LP_DEPOSITORY_NAMESPACE,
+        credix_global_market_state.as_ref(),
+        collateral_mint.as_ref(),
+        &[ctx.accounts.depository.load()?.bump],
+    ]];
+
+    // ---------------------------------------------------------------------
+    // -- Phase 1
+    // -- Check if the withdraw redeem period is active at the moment
+    // ---------------------------------------------------------------------
+
+    let start_of_redeem_phase_timestamp = ctx
+        .accounts
+        .credix_withdraw_epoch
+        .go_live
+        .checked_add(ctx.accounts.credix_withdraw_epoch.redeem_seconds.into())
+        .ok_or(UxdError::MathError)?;
+    let end_of_redeem_phase_timestamp = start_of_redeem_phase_timestamp
+        .checked_add(ctx.accounts.credix_withdraw_epoch.redeem_seconds.into())
+        .ok_or(UxdError::MathError)?;
+    require!(
+        (start_of_redeem_phase_timestamp..end_of_redeem_phase_timestamp)
+            .contains(&Clock::get()?.unix_timestamp),
+        UxdError::InvalidCredixWithdrawEpochRequestPeriod,
+    );
+
+    // ---------------------------------------------------------------------
+    // -- Phase 2
+    // -- Fetch all current onchain state
+    // ---------------------------------------------------------------------
+
+    let credix_lp_depository_collateral_amount_before: u64 =
+        ctx.accounts.depository_collateral.amount;
+    let identity_depository_collateral_amount_before: u64 =
+        ctx.accounts.identity_depository_collateral.amount;
+    let profits_beneficiary_collateral_amount_before: u64 =
+        ctx.accounts.profits_beneficiary_collateral.amount;
+
+    let liquidity_collateral_amount_before: u64 = ctx.accounts.credix_liquidity_collateral.amount;
+    let outstanding_collateral_amount_before: u64 = ctx
+        .accounts
+        .credix_global_market_state
+        .pool_outstanding_credit;
+
+    let total_shares_supply_before: u64 = ctx.accounts.credix_shares_mint.supply;
+    let total_shares_value_before: u64 = liquidity_collateral_amount_before
+        .checked_add(outstanding_collateral_amount_before)
+        .ok_or(UxdError::MathError)?;
+
+    let owned_shares_amount_before: u64 = ctx.accounts.depository_shares.amount;
+    let owned_shares_value_before: u64 = compute_value_for_shares_amount_floor(
+        owned_shares_amount_before,
+        total_shares_supply_before,
+        total_shares_value_before,
+    )?;
```

Comment:
nit: scope variables used only locally

```suggestion
    let credix_lp_depository_collateral_amount_before: u64 =
        ctx.accounts.depository_collateral.amount;
    let identity_depository_collateral_amount_before: u64 =
        ctx.accounts.identity_depository_collateral.amount;
    let profits_beneficiary_collateral_amount_before: u64 =
        ctx.accounts.profits_beneficiary_collateral.amount;

    let total_shares_supply_before: u64 = ctx.accounts.credix_shares_mint.supply;
    let total_shares_value_before: u64 = {
        let liquidity_collateral_amount_before: u64 =
            ctx.accounts.credix_liquidity_collateral.amount;
        let outstanding_collateral_amount_before: u64 = ctx
            .accounts
            .credix_global_market_state
            .pool_outstanding_credit;

        liquidity_collateral_amount_before
            .checked_add(outstanding_collateral_amount_before)
            .ok_or(UxdError::MathError)?
    };

    let owned_shares_amount_before: u64 = ctx.accounts.depository_shares.amount;
    let owned_shares_value_before: u64 = compute_value_for_shares_amount_floor(
        owned_shares_amount_before,
        total_shares_supply_before,
        total_shares_value_before,
    )?;
```

Reply:
Commited and applied for the before+after